### PR TITLE
fix: ci lint for mod registry and image

### PR DESCRIPTION
**Pull request by justadogistaken**
do golangci-lint on registry&image module,  and fixed issues occurred

### CODE REVIEW

File: image/default_image.go [hunk 1]
```diff
@@ -309,7 +309,9 @@ func (d DefaultImageService) uploadLayers(repo string, layers []v1.Layer, blobs
 						errCh <- err
 						return err
 					}
-					defer utils.CleanFile(file)
+					defer func() {
+						_ = utils.CleanFile(file)
```
Comment (fanux) [inline image/default_image.go:313]: better make a warn log

**Comment by fanux** [comment]
should not ignore errors! @justadogistaken

File: image/default_image.go (hunk 1 above)
Comment (justadogistaken) [reply image/default_image.go:313]: agree on this

**Review by fanux — APPROVED** [review]
handle errors

File: utils/file.go [hunk 2]
```diff
@@ -146,12 +146,20 @@ func CopySingleFile(src, dst string) (int64, error) {
...
-	err = file.Close()
-	return os.Remove(file.Name())
+	// the following operation won't failed regularly, if failed, log it
+	err := file.Close()
+	if err != nil {
+		logger.Warn(err)
+	}
+	err = os.Remove(file.Name())
+	if err != nil {
+		logger.Warn(err)
```
Comment (fanux) [inline utils/file.go:160]: The recommended way is to return the error, caller decision write log or not

Comment (justadogistaken) [reply utils/file.go:160]: I suppose os.Remove/Close ... won't return err regularly. So if error occurs, we log it.

File: image/util.go [hunk 3]
```diff
@@ -90,7 +90,9 @@ func GetClusterFileFromImageManifest(imageName string) string {
 func GetClusterFileFromBaseImage(imageName string) string {
 	mountTarget, _ := utils.MkTmpdir()
 	mountUpper, _ := utils.MkTmpdir()
-	defer utils.CleanDirs(mountTarget, mountUpper)
+	defer func() {
+		_ = utils.CleanDirs(mountTarget, mountUpper)
```
Comment (fanux) [inline image/util.go:94]: warn log

**Pull request merged**